// Copyright (c) 2021 Huawei Device Co., Ltd.
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
syntax = "proto3";
option java_package = "ohos.devtools.datasources.transport.grpc.service";
option optimize_for = LITE_RUNTIME;

enum Type {
    TYPE_UNSPECIFIED = 0;
    HI3516 = 1;
    P40 = 2;
}

enum Level {
    LEVEL_UNSPECIFIED = 0;
    ERROR = 1;
    INFO = 2;
    DEBUG = 3;
    WARN = 4;
}

message HilogConfig {
    Type device_type = 1;
    Level log_level = 2;

    // set the pid for get log e.g. hilogcat --pid=111
    int32 pid = 3;
    bool need_record = 4;
    bool need_clear = 5;
}